crates/ra_ide_api/src/join_lines.rs

```diff
@@ -9,9 +9,8 @@ use ra_syntax::{
 use ra_fmt::{
     compute_ws, extract_trivial_expression
 };
-use crate::{
-    LocalEdit, TextEditBuilder,
-};
+use ra_text_edit::TextEditBuilder;
+use ra_ide_api_light::LocalEdit;
```

Comment:
It should be possible to avoid using `LocalEdit` here and switch to `SourceFileEdit` directly

Reply:
That is, the *existence* of `LocalEdit` is just an artifact of the split between light/non-light API. If we remove the split, `LocalEdit` become unnecessary, as it is equivalent to SourceFileEdit/SourceEdit

Reply:
Since `LocalEdit` is still used from `typing.rs`, it's probably easier to first move that and then remove `LocalEdit` in one go?

Reply:
Oh right, it's independent enough to be removed here.